clients/src/main/java/org/apache/kafka/common/utils/KafkaThread.java

```diff
@@ -28,19 +28,19 @@
 
     public KafkaThread(final String name, boolean daemon) {
         super(name);
-        configureThread(name, daemon);
+        configureThread(daemon);
     }
 
     public KafkaThread(final String name, Runnable runnable, boolean daemon) {
         super(runnable, name);
-        configureThread(name, daemon);
+        configureThread(daemon);
     }
 
-    private void configureThread(final String name, boolean daemon) {
+    private void configureThread(boolean daemon) {
```

Comment:
I think you still pass the name directly. Getting the name from the superclass involves a volatile read and it's unnecessary since we have it here.